bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/mirroring/utils/DatabaseHelpers.java

```diff
@@ -0,0 +1,186 @@
+
+  public DatabaseHelpers(ConnectionRule connectionRule, ExecutorServiceRule executorServiceRule) {
+    this.executorServiceRule = executorServiceRule;
+    this.connectionRule = connectionRule;
+  }
+
+  public enum DatabaseSelector {
+    PRIMARY,
+    SECONDARY
+  }
+
+  public int countCells(TableName tableName, DatabaseSelector databaseSelector) throws IOException {
+    return countRowsCells(tableName, databaseSelector, new Scan()).cells;
+  }
+
+  public int countRows(TableName tableName, DatabaseSelector databaseSelector) throws IOException {
+    return countRowsCells(tableName, databaseSelector, new Scan()).rows;
+  }
+
+  public int countRows(TableName tableName, DatabaseSelector databaseSelector, Scan scan)
+      throws IOException {
+    return countRowsCells(tableName, databaseSelector, scan).rows;
+  }
+
+  public static class RowCellsCount {
+    public final int rows;
+    public final int cells;
+
+    public RowCellsCount(int rows, int cells) {
+      this.rows = rows;
+      this.cells = cells;
+    }
+  }
+
+  public RowCellsCount countRowsCells(
+      TableName tableName, DatabaseSelector databaseSelector, Scan scan) throws IOException {
+    int rowsCounter = 0;
+    int cellsCounter = 0;
+    try (MirroringConnection mirroringConnection = createConnection()) {
+      Connection connection = selectConnection(mirroringConnection, databaseSelector);
+      try (Table t1 = connection.getTable(tableName)) {
+        try (ResultScanner scanner = t1.getScanner(scan)) {
+          for (Result r : scanner) {
+            rowsCounter++;
+            CellScanner cellScanner = r.cellScanner();
+            do {
+              if (cellScanner.current() != null) {
+                cellsCounter++;
+              }
+            } while (cellScanner.advance());
+          }
+        }
+      }
+    }
+    return new RowCellsCount(rowsCounter, cellsCounter);
+  }
+
+  public Connection selectConnection(
+      MirroringConnection mirroringConnection, DatabaseSelector databaseSelector) {
+    switch (databaseSelector) {
+      case PRIMARY:
+        return mirroringConnection.getPrimaryConnection();
+      case SECONDARY:
+        return mirroringConnection.getSecondaryConnection();
+      default:
+        throw new UnsupportedOperationException("Invalid database selector");
+    }
+  }
+
+  public void fillTable(
+      TableName tableName1, int numberOfRows, byte[] columnFamily, byte[]... qualifiers)
+      throws IOException {
+    try (MirroringConnection connection = createConnection()) {
+      try (Table t1 = connection.getTable(tableName1)) {
+        for (int id = 0; id < numberOfRows; ) {
+          List<Put> puts = new ArrayList<>();
+          for (int j = 0; j < 100 && id < numberOfRows; j++) {
+            byte[] rowKey = Longs.toByteArray(id);
+            Put put = new Put(rowKey);
+            for (byte[] qualifier : qualifiers) {
+              put.addColumn(columnFamily, qualifier, id, rowKey);
```

Comment:
Changing this would be painful, we are using this assumption in a few places. But I don't think that it is a problem because there are tests that use different rowkey and value and they would fail if we swapped them.

Reply:
Ack !